File: src/main/java/my/maryvk/maryvkweb/seeker/FriendsAndFollowersMarySeeker.java
package my.maryvk.maryvkweb.seeker;

import my.maryvk.maryvkweb.domain.RelationType;
import my.maryvk.maryvkweb.service.RelationService;
import my.maryvk.maryvkweb.service.VkService;

public class FriendsAndFollowersMarySeeker implements MarySeeker {

    private final MarySeeker friendsMarySeeker;
    private final MarySeeker followersMarySeeker;

    private final int usedId;

    public FriendsAndFollowersMarySeeker(VkService vk, int userId, RelationService relationService) {
        friendsMarySeeker = new MarySeekerImpl(vk, userId, RelationType.FRIEND, relationService);
        followersMarySeeker = new MarySeekerImpl(vk, userId, RelationType.FOLLOWER, relationService);
        this.usedId = userId;
    }

    @Override
    public void seek() {
        friendsMarySeeker.seek();
        followersMarySeeker.seek();
    }

    @Override
    public int getUserId() {
        return usedId;
    }
}

File: src/main/java/my/maryvk/maryvkweb/seeker/FriendsAndFollowersMarySeeker.kt
package my.maryvk.maryvkweb.seeker

import my.maryvk.maryvkweb.domain.RelationType
import my.maryvk.maryvkweb.service.RelationService
import my.maryvk.maryvkweb.service.VkService

class FriendsAndFollowersMarySeeker(vk: VkService, override val userId: Int, relationService: RelationService) : MarySeeker {

    private val friendsMarySeeker: MarySeeker
    private val followersMarySeeker: MarySeeker

    init {
        this.friendsMarySeeker = MarySeekerImpl(vk = vk, userId = userId, relationType = RelationType.FRIEND, relationService = relationService)
        this.followersMarySeeker = MarySeekerImpl(vk = vk, userId = userId, relationType = RelationType.FOLLOWER, relationService = relationService)
    }

    override fun seek() {
        friendsMarySeeker.seek()
        followersMarySeeker.seek()
    }
}